Adds solution to problem 1 of arrays

tahmidsadik opened the pull request.

tahmidsadik committed:
Adds solution to problem 1 of arrays

mugli reviewed (commented):
I have two suggestions for improvement 🙏

- Prettier 💅 🙃
- Could we also include a runnable block at the end of the script that verifies the code is working as expected, [like this](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEB6AVOgBAHSl7ABQCcIAjAGzgFssBnGAQ3moRl32wBUB3CLAHIBXamTjEsAZRF4CHOQHViAS3hZGWAGZCoYGMuhYYAC2ZHGAazh0sy-BqjQAtDQAOMAJ7rixRl4iaWAAmygx2erawcADm4jaMUEHqULLYdvCxEsRwrtl0bHbR6ubEsex0IgB0WACSgQleMHxYUCJixDZ2RsZwka5C7Iw+fvQynFhCrkb8Jr1MpXDlIgA03b3auvqGdMYQQhRJ2TBCxPiztF0J3r4eq5dQHqlYEMRB4tV1LTPNraJxo7RJtM1iUygDVrMtDo9AZ8Ds9gcsEcTilxlc3J5rn5KqkngIIGoTGZIfMwRVaKYbDB+GJnmQmHY4EkyF5ydRCkZmiFNJpxGxIhk4k8upChjcANxYDx7LBgBIAckGQSSGjohSoApi4mBqnyFECXRebwk1LpDLOPSepMWAJxnCeAE0ZXL7HRyXNTOxZtksDxlBQKFhaWZqBAGM8oL1XIxlBIAi02v82RygdSrT1QTbyXaCE9JIxqK4NTUoP12E8xSMALxYADaAGZVgBWVZOAAsqwAHKsAIw93ut-tYABsAF0rUMytJaDWewAGXHjfOFjUAeQGZaetacQ774-t43QqDweA2MMMTQgwj+xGnAApK7dMzBpwBKLDAE8wSBQcO-doALKMFMNbAAAvuKX6aC8WB3lQ7DKM89TDB476fikMA4DA-7iEBri1soo5YLOkEYWBUEvHeP7hmAJzZLAABqjCBvGj5oV+WHUewFAQBAFgAGIwTW1oviIWBOLKdFsExFCkVg8lYKgqC+r0kZMsCYA9GAFjdMSGa0T40nMckSSQjhJr8DxfFaDBQwIIqsgKUpIJ0AWcCrGIcpCPkILmZ0Fq9I+3i9CEYSbNUHH6Jod6SYZjHGQAhFWNZWQJMEAGTpQmN54bWqWCcQo7sRhWFYcipx1gZ9EwDJqz5S8o6kaVMDkWRJ4YeV+C1o1eCtXgzkKD09haP6vShEiOjBLGcB6BQXiaKQtD4m8qzZAAjkI02VNQMb4ON+RevwupYKGQT7HAdrKIEd7rZt2TbbtxHJSdEBnVQxXyVx6huuI7A1rd013vKjA-cQMDyq+lQMCoehyfIsrQOGdhlsRdaNlgLbiR2WDdlgfYDuJQ5jnDX0idOqPznDjlfXAAAergzfAQTrjAKM1tuu49j1+AI7+7B7KzAyo5e17tPeyMDBCk6LG+VM8yD+Rg5Ubw5JIMAwzAACiG3MXedMM3oTIs2WUMvDAd6vqsAsm3QZsW6+pFgSAywgBA7iwnQyCgGKEA8IQdm-sgIDMTwfiey7ZC+NpMvRmAhTIOrQjuSAxgwNQFCDao1ix3AkgIGq+gAG6qB4QdgG6zsgHYiswCQjDRDtyCaMx+QuwAVnQtMAEJR1YL5uQAMoyTct8nHe05I6pwAAikIBJwCPFCtyA0YdOIQdkIwYgUJXuTpAoyhBCYyCdnOLu5GGcBKMBQe5NY4iFwvLsbfPJBu57KAg04alvEElcA9kOuDdGCL2XvkdkCdiBJxdmqKA0QqCz3nqA5OTAyAHyPsYZAAAmF26sYwUEKAAYQgNQRuKBrBNkrt5OAXAt4f2bkvZOhck4ljeLASQYAVDuAAIKJDVh4KgyCwJgSAA)? One simple test case is enough just to make sure it's working as  expected. 

In case there are multiple implementations with varying efficiency, we will be able to verify that all of them are producing the same output.

tahmidsadik committed:
Updates solution for two-number-sum problem

- Adds a folder for the problem.
- Renamed the p1.js file to solution.js.
- Adds a readme file containing problem statement, sample input and
  output, time complexity and
  space complexity.
- Adds simple test assertion at the end of the solution file.
- Updated the solution to match time/space complexity mentioned.

tahmidsadik commented:
I liked the way you have structured your projected ( folder structure, file naming ), I have updated mine according to that. Added simple test case for the problem, separate read-me file with meta information. 

I have also updated the solution - my previous solution had worse time complexity( probably O(n^2) ) - the worst case for the updated solution should be O(n) for both time and space complexity.
